debezium-core/src/main/java/io/debezium/relational/history/KafkaDatabaseHistory.java

```diff
@@ -224,16 +221,25 @@ protected void recoverRecords(Tables schema, DdlParser ddlParser, Consumer<Histo
                     }
                 }
                 if (numRecordsProcessed == 0) {
-                    --remainingEmptyPollResults;
-                    logger.debug("No new records found in the database history; will retry {} more times", remainingEmptyPollResults);
+                    logger.debug("No new records found in the database history; will retry");
                 } else {
-                    remainingEmptyPollResults = this.recoveryAttempts;
                     logger.debug("Processed {} records from database history", numRecordsProcessed);
                 }
             }
         }
     }
 
+    private long getEndOffsetOfDbHistoryTopic(KafkaConsumer<String, String> historyConsumer) {
+        Map<TopicPartition, Long> offsets = historyConsumer.endOffsets(Collections.singleton(new TopicPartition(topicName, PARTITION)));
+
+        if (offsets.isEmpty()) {
+            return Long.MAX_VALUE;
```

Comment:
Won't this cause the consumer logic to run forever? Should we instead throw an exception if we can't get the offset for our topic partition? OTOH, if that does happen before the topic is created, then it seems like `-1` might be a better return value.

Reply:
Indeed. It seems that's actually never happening, though, as I'll get back a map with 0 as the end offset in the initial case.

Reply:
If I understand correctly, `offsets.isEmpty()` means the topic doesn't exist and this relates somewhat to https://issues.jboss.org/browse/DBZ-443 

FWIW, as part of that story I was planning on adding an `exists()` method to DatabaseHistory interface. So when implemented in KafkaDatabaseHistory something like:
```java
    @Override
    public boolean exists() {
    	boolean exists = false;
    	try (KafkaConsumer<String, String> historyConsumer = new KafkaConsumer<>(consumerConfig.asProperties());) {
    		exists = !historyConsumer.endOffsets(Collections.singleton(new TopicPartition(topicName, PARTITION))).isEmpty();  	
    	}
    	return exists;
    }
```
This would be called before we try to call loadHistory during task startup, so we can never get to the scenario trying to read a history topic that does not exist. That would likely negate the need for this additional check in `getEndOffsetOfDbHistoryTopic` ?

Reply:
That sounds good. In fact I'm never seeing that it returns an empty map, when testing with a non-existing topic, it'll still return 0 for the requested partition. So I think the implementation is sound as of my last rework, but if you could check it as you suggest, that'd be very good.